text/1000-migrations.rst

```diff
@@ -40,6 +40,10 @@ the necessary transformations using DDL and, possibly, regular queries.
 In the database, the migrations are recorded as ``sys::Migration`` objects::
 
     type sys::Migration {
+        # Migration hash derived from contents and parent hashes.
+        required property hash -> str {
+            constraint exclusive;
+        };
```

Comment:
I don't see the `name` property and the discussion on how migration names would be derived from hashes. Also how one uses the `DESCRIBE` or `DESCRIBE MIGRATION` commands with migration hashes.

Reply:
Do we still want to have separate "hash" and "name" fields?  Arbitrary migration names?

Reply:
I'd be fine with `DESCRIBE MIGRATION "hash"` variant and not having `name` field for migrations at all. But I think you wanted them to have names, no?

Reply:
OK, I think we should just use the `name` property, which is generated using the hash.

Reply:
Hm, I'd keep the hash as a separate property. Maybe we define `name` as a required property with a computable default off the hash?

Reply:
Hm, I think this arrangement makes things more confusing than not.  We refer to parent migrations by hash, but use "name" everywhere else?  I think we should stick with a single identifying property and disallow anything but a hash-derived value in it.

Reply:
The name essentially becomes hash, hence don't expect that we ever be able to have custom names.

Reply:
I also don't like that names essentially invites people to circumvent the hashing of migrations and the integrity checks that hashing enables. I really don't like this whole "hash" is a "name" business.

Reply:
> I also don't like that names essentially invites people to circumvent the hashing of migrations

I changed my mind re custom names.  They're unnecessary and complicate things, because now you how TWO ways to identify and refer to a migration (a hash and a name).  I think we should settle on one, system-maintained.  Calling it "name" makes things much simpler for the implementation, because all other objects are identified by "name" and lots of code is written with this assumption.

Reply:
> I changed my mind re custom names.

Good.

I still don't like the proposed schema design though. It creates an impression that names *can be custom*. Can we do this:

```
type sys::Migration {
        # Migration name derived from the hash of contents and parent name.
        required property hash -> str {
            constraint exclusive;
        };
        required property name  := .hash;
}
```

?

Reply:
This arrangement would require hacks on the reflection side, because internally we would still need to define the field as "name", otherwise we would not be able to rely on the common infrastructure for "named" schema objects, as I pointed out above.  This seems like an unnecessary price to pay for me.

Reply:
Would it even be possible to set `name` for the user somehow? I.e. `CREATE MIGRATION` and `COMMIT MIGRATION` will compute the hash/name automatically and there's no need for `SET name := ...` syntax, correct?

Reply:
Yes, no need, it's read-only from the user's perspective.

Reply:
OK then, case closed.